### PR TITLE
Bau ser 02 upload azure

### PR DESCRIPTION
**Pull request by kduy**

### CODE REVIEW

File: app/controllers/api/v1/claims_controller.rb [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+  def create
+    @claim = Claim.new(claim_params)
+
+    @claim.save
+    render json: @claim,
+           serializer: ClaimSerializer,
+           root: 'claim'
+  end
+
+  def edit
```
Comment (vsrboth) [inline app/controllers/api/v1/claims_controller.rb]: please remove this. We don't need this.

File: app/controllers/api/v1/claims_controller.rb [hunk 2]
```diff
@@ -0,0 +1,52 @@
+class Api::V1::ClaimsController < ApplicationController
+  before_action :set_claim, except: [:create]
+  def show
```
Comment (vsrboth) [inline app/controllers/api/v1/claims_controller.rb]: there should be an api to filter and also get all claims.

for getting all claims, plz use **pagination** and we should return only 10 claims per page.

File: app/models/claim_photo.rb [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+#
+#  index_claim_photos_on_claim_id  (claim_id)
+#
+
+class ClaimPhoto < ApplicationRecord
+  after_initialize :defaults, unless: :persisted?
+
+  belongs_to :claim
+  has_attached_file :photo
+  validates_attachment_content_type :photo, content_type: %w[image/jpg, image/jpeg, image/png, image/gif]
```
Comment (vsrboth) [inline app/models/claim_photo.rb]: %w() not %w[]

File: app/controllers/api/v1/claims_controller.rb (hunk 1 above)
Comment (kduy) [inline app/controllers/api/v1/claims_controller.rb]: remove?

File: app/models/errors.rb [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+    {:href => href, :method => method, :rel => "request"}
+  end
+
+  def entry
+    href = URI::encode("/api/v1/")
+    {:href => href, :method => "GET", :rel => "entry point", :params => params}
+  end
+
+  def params
+    {:point => {:value => "{lon},{lat}", :optional => true}, :title => {:value => "text", :optional => true}}
```
Comment (vsrboth) [inline app/models/errors.rb]: prefer this way:

```ruby
{
  point: {
    value: "#{lon}, #{lat}",
    optional: true,
    title: {
      value: 'text',
      optional: true
    }
  }
}
```

File: app/models/errors.rb [hunk 5]
```diff
@@ -0,0 +1,41 @@
+# https://github.com/hiromipaw/citywalks/blob/8eab6af6d6265e7314b3a710c2ff4583577946bc/app/models/error.rb
+class Errors
+  include ActiveModel::Validations
+  include ActiveModel::Conversion
+  extend ActiveModel::Naming
+
+  attr_accessor :text, :status, :url, :method
+
+  def _links
+    {:url => request, :entry => entry}
```
Comment (vsrboth) [inline app/models/errors.rb]: prefer this style: `url: request` to `:url => request`

File: app/models/errors.rb (hunk 4 above)
Comment (kduy) [inline app/models/errors.rb]: done

File: app/models/errors.rb (hunk 5 above)
Comment (kduy) [inline app/models/errors.rb]: done

File: app/models/claim_photo.rb (hunk 3 above)
Comment (kduy) [inline app/models/claim_photo.rb]: done

File: app/controllers/api/v1/claim_photos_controller.rb [hunk 6]
```diff
@@ -0,0 +1,54 @@
+class Api::V1::ClaimPhotosController < ApplicationController
+  before_action :set_claim_photo, except: [:create, :index]
+
+  def index
+    if params[:page].blank?
+      pageNo = 1
+    else
+      pageNo = params[:page]
```
Comment (vsrboth) [inline app/controllers/api/v1/claim_photos_controller.rb]: plz do not use `pageNo`, prefer `page_no`

Comment (vsrboth) [inline app/controllers/api/v1/claim_photos_controller.rb]: default page is 1, please do not check whether it is blank or not.

**Pull request merged**